include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort.h

```diff
@@ -739,19 +739,19 @@ __parallel_radix_sort_iteration(_ExecutionPolicy&& __exec, ::std::size_t __segme
     using _AscendingType = ::std::bool_constant<__is_comp_asc>;
     using _CustomName = typename __decay_t<_ExecutionPolicy>::kernel_name;
     using _RadixCountKernel =
-        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__radix_sort_count_kernel, _CustomName,
-                                                                               _RadixBitsType, _AscendingType,
+        oneapi::dpl::__par_backend_hetero::__internal::__kernel_name_generator<__radix_sort_count_kernel, __radix_sort_count_kernel<>,
+                                                                               _CustomName, _RadixBitsType, _AscendingType,
```

Comment:
As we discussed in person, let's not pass the same name twice to the generator, but inside it use the BaseName twice, with empty pack and with the provided types.